addons/l10n_in/models/res_partner.py

```diff
@@ -72,3 +72,28 @@ def check_vat_in(self, vat):
         if vat == TEST_GST_NUMBER:
             return True
         return super().check_vat_in(vat)
+
+    @api.model
+    def _l10n_in_get_partner_vals_by_vat(self, vat):
+        partner_details = self.read_by_vat(vat)
+        partner_data = partner_details[0] if partner_details else {}
+        if partner_data:
+            partner_gid = partner_data.get('partner_gid')
+            if partner_gid:
+                partner_data = self.enrich_company(company_domain=None, partner_gid=partner_gid, vat=partner_data.get('vat'))
+                partner_data = self._iap_replace_logo(partner_data)
+            return {
+                'name': partner_data.get('name'),
+                'company_type': 'company',
+                'partner_gid': partner_gid,
+                'vat': partner_data.get('vat'),
+                'l10n_in_gst_treatment': 'regular',
+                'image_1920': partner_data.get('image_1920'),
+                'street': partner_data.get('street'),
+                'street2': partner_data.get('street2'),
+                'city': partner_data.get('city'),
+                'state_id': partner_data.get('state_id', False),
+                'country_id': partner_data.get('country_id', False),
```

Comment:
Do we need the `False` default value here? Isn't `None` from `get` enough if we're just using this dict in `.create(...)` ?

Reply:
I will keep my the code back:

```python
'state_id': partner_data.get('state_id') and partner_data.get('state_id').get('id') or False,
```

The `read_by_vat` method returns `country_id` and `state_id` as dictionaries that include both `id` and `display_name`. Therefore, it's necessary to set the value to `False` if no data is retrieved.

For reference, you can check the code here where the values are set: Code Reference https://github.com/odoo/odoo/blob/5ad3315b260206be564c6ea08328d237bc656f5f/addons/partner_autocomplete/models/res_partner.py#L56

Reply:
I see, then let's improve it with my first suggestion then.
`'state_id': partner_data.get('state_id', {}).get('id', False)`

proof in python:
```py
>>> a = {'c': {'id': 123}}
>>> b = {}
>>> a.get('c', {}).get('id', False)
123
>>> b.get('c', {}).get('id', False)
False

Reply:
Hello @yosuanicolaus ,
I have made the changes.